soroban-env-common/src/error.rs

```diff
@@ -163,6 +163,14 @@ impl From<stellar_xdr::Error> for Error {
     }
 }
 
+// This never happens, but it's needed for some impls of TryFromVal downstream
+// in the SDK that use the xdr::Error type.
+impl From<Error> for stellar_xdr::Error {
+    fn from(_value: Error) -> Self {
+        stellar_xdr::Error::Unsupported
```

Comment:
If this never happens should it be `unreachable!()`?

Reply:
I _think_ it never happens but I don't think it's logically impossible, and I'm trying to avoid anything that can panic when there's a bug somewhere.